buildtest/buildsystem/base.py

```diff
@@ -82,15 +82,14 @@ def load(self, buildspec):
            :param buildspec: the pull path to the configuration file, must exist.
            :type buildspec: str, required
         """
-        self.buildspec = os.path.abspath(buildspec)
 
-        if not is_file(self.buildspec):
-            sys.exit("Buildspec File: %s does not exist." % self.buildspec)
+        self.buildspec = resolve_path(buildspec)
```

Comment:
The check if buildspec is None needs to go _before_ this line - if buildspec here is None, it's going to spit out an error when you do expandvars in resolve_path.

Reply:
E.g.,
import os                                                                                                 
# This is the first line of resolve_path
path = os.path.expandvars(None)                                                                           
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-4154076a32c3> in <module>
----> 1 path = os.path.expandvars(None)

~/anaconda3/lib/python3.7/posixpath.py in expandvars(path)
    286     """Expand shell variables of form $var and ${var}.  Unknown variables
    287     are left unchanged."""
--> 288     path = os.fspath(path)
    289     global _varprog, _varprogb
    290     if isinstance(path, bytes):

TypeError: expected str, bytes or os.PathLike object, not NoneType
```

Reply:
Okay i made the change and tested this locally to ensure it hits all three conditionals. 

```
>>> spec = BuildspecParser(None)
Invalid input type for Buildspec, must be of type 'string'.
>>> spec = BuildspecParser("$PWD/invalid/file")
Can't process input: $PWD/invalid/file
>>> spec = BuildspecParser("$HOME/.buildtest")
Detected /u/users/ssi29/.buildtest is a directory, please provide a file path (not a directory path) to BuildspecParser.

```

Reply:
@vsoch  i also added two more checks in ``test_base.py`` in commit https://github.com/shahzebsiddiqui/buildtest-framework/commit/b37a93a98f38f6e7728dca154a834bd5b7b7d93f to make sure we hit those conditionals in coverage report.